src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -181,62 +187,62 @@ class CosmasGoogleCloudService(
         val response = CosmasProto.CommitVersionResponse.newBuilder()
         synchronized(project) {
             try {
-                val prevIds = getPrevIds(request.info).toMutableMap()
-                for ((fileId, fileVersion) in project.asMap()) {
-                    try {
-                        MDC.clear()
-                        MDC.put("fileId", fileId)
-                        val text = fileVersion.content.toStringUtf8()
-                        val patches = fileVersion.patchesList.toMutableList()
-                        if (patches.isEmpty()) {
-
-                            LOG.info("File has no patches, no need to commit it")
-                            continue
-                        }
-                        patches.sortBy { it.timestamp }
-
-                        val newText = PatchCorrector.applyPatch(patches, text)
-                        val cosmasHash = md5Hash(newText)
-                        if (patches.last().actualHash == cosmasHash) {
-                            commitFromMemoryToGCS(request.info, fileId, ByteString.copyFromUtf8(newText), null)
-                            LOG.info("File has been committed")
-                        } else {
-                            val actualHash = patches.last().actualHash
-                            LOG.error("""Commit failure: hash mismatch.
+                mediator.withWriteFileIdChangeMap(request.info) { fileIdChangeMap ->
+                    val prevIds = fileIdChangeMap.prevIdsMap.toMutableMap()
+                    for ((fileId, fileVersion) in project.asMap()) {
+                        try {
+                            MDC.clear()
+                            MDC.put("fileId", fileId)
+                            val text = fileVersion.content.toStringUtf8()
+                            val patches = fileVersion.patchesList.toMutableList()
+                            if (patches.isEmpty()) {
+
+                                LOG.info("File has no patches, no need to commit it")
+                                continue
+                            }
+                            patches.sortBy { it.timestamp }
+
+                            val newText = PatchCorrector.applyPatch(patches, text)
+                            val cosmasHash = md5Hash(newText)
+                            if (patches.last().actualHash == cosmasHash) {
+                                commitFromMemoryToGCS(request.info, fileId, ByteString.copyFromUtf8(newText), null)
+                                LOG.info("File has been committed")
+                                var curFileId = fileId
```

Comment:
This loop requires some comments, like what's going on here.
My version:
```
We keep a history of file id changes in `fileIdChangeMap`. 
We need it only until the first write of FileVersion to persistent storage. 
After the write previous ids will be available from .... records. 
Thus, we clean up the file id change mapping by removing records corresponding to written file.
```

Reply:
ok